src/translator/service.cpp

```diff
@@ -40,21 +63,43 @@ Service::Service(Ptr<Options> options, MemoryGift modelMemory, MemoryGift shortl
       pcqueue_(std::max<size_t>(1, numWorkers_))
 #endif
 {
+  initialize(options);
+}
 
-  if (numWorkers_ == 0) {
-    build_translators(options, /*numTranslators=*/1);
-    initialize_blocking_translator();
-  } else {
-    build_translators(options, numWorkers_);
-    initialize_async_translators();
+Service::Service(const std::string &config, const void* modelMemory, const void* shortlistMemory)
+    : numWorkers_(parseOptions(config)->get<int>("cpu-threads")),
+      text_processor_(vocabs_, parseOptions(config)), batcher_(parseOptions(config))
+#ifndef WASM
+      // 0 elements in PCQueue is illegal and can lead to failures. Adding a
+      // guard to have at least one entry allocated. In the single-threaded
+      // case, while initialized pcqueue_ remains unused.
+      ,
+      pcqueue_(std::max<size_t>(1, numWorkers_))
+#endif
+{
+  Ptr<Options> options = parseOptions(config);
+  initialize(options);
```

Comment:
if initialize -> build_translators (which passes the modelMemory_), the code below this becomes dead code right?